### PR TITLE
includes service resource usage in /apps endpoints

### PR DESCRIPTION
**Pull request by shamsimam**
## Changes proposed in this PR

- includes service resource usage in /apps endpoints

## Why are we making these changes?

Avoid the need for clients to compute this value.

Screen Shot 2020-04-21 at 1 08 31 PM
Screen Shot 2020-04-21 at 1 08 20 PM

### CODE REVIEW

File: waiter/src/waiter/handler.clj [hunk 1]
```diff
@@ -330,15 +336,19 @@
...
                                  :service-id service-id
-                                 :service-description service-description
+                                 :service-description core-service-description
                                  :status (service/retrieve-service-status-label service-id global-state)
-                                 :url (prepend-waiter-url (str "/apps/" service-id))}
+                                 :url (prepend-waiter-url (str "/apps/" service-id))
+                                 :usage usage-map}
                                 include-effective-parameters?
                                 (assoc :effective-parameters
                                        (service-id->service-description-fn service-id :effective? true))
```
Comment (sradack) [inline waiter/src/waiter/handler.clj:354]: We're doing this twice.

File: waiter/src/waiter/handler.clj [hunk 2]
```diff
@@ -330,15 +336,19 @@
...
-                                {:instance-counts (retrieve-instance-counts service-id)
+                                {:instance-counts instance-counts
                                  :last-request-time (get-in service-id->metrics [service-id "last-request-time"])
                                  :service-id service-id
-                                 :service-description service-description
+                                 :service-description core-service-description
                                  :status (service/retrieve-service-status-label service-id global-state)
-                                 :url (prepend-waiter-url (str "/apps/" service-id))}
+                                 :url (prepend-waiter-url (str "/apps/" service-id))
+                                 :usage usage-map}
```
Comment (sradack) [inline waiter/src/waiter/handler.clj:351]: Nitpick: I think you're right.  `:resource-usage` feels better to me.

Comment (sradack) [reply waiter/src/waiter/handler.clj:351]: I considered naming this `:resources-requested` (or something similar) and and reserve `:resource-usage` to mean actual usage.

Comment (sradack) [reply waiter/src/waiter/handler.clj:351]: Ultimately decided `:resources-requested` feels awfully k8s-specific.

File: waiter/src/waiter/handler.clj (hunk 1 above)
Comment (shamsimam) [reply waiter/src/waiter/handler.clj]: Removed duplicate computation.

File: waiter/src/waiter/handler.clj (hunk 2 above)
Comment (shamsimam) [reply waiter/src/waiter/handler.clj]: Renamed to `resource-usage`

**Pull request merged**